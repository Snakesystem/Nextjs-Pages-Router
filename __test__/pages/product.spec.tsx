import ProductPage from "@/pages/product";
import { render } from "@testing-library/react";

jest.mock("next/router", () => {
    return {
        useRouter() {
            return { 
                route: "/product",
                pathname: "",
                query: "",
                asPath: "",
                push: jest.fn(),
                event: {
                    on: jest.fn(),
                    off: jest.fn()
                },
                beforePopState: jest.fn(() => null),
                prefetch: jest.fn(() => null),
                isReady: true
            }
        },
    }
});

describe("Product Page", () => {
    it("render product page", () => {
        const page = render(<ProductPage />);
        expect(page).toMatchSnapshot();
    });
})