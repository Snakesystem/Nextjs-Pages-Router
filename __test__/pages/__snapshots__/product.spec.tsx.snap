// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Page render product page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <div
          class="product"
        >
          <h1
            class="product__title"
          >
            Product
          </h1>
          <div
            class="product__content"
          >
            <div
              class="product__content__skeleton"
            >
              <div
                class="product__content__skeleton__image"
              />
              <div
                class="product__content__skeleton__name"
              />
              <div
                class="product__content__skeleton__category"
              />
              <div
                class="product__content__skeleton__price"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <div
        class="product"
      >
        <h1
          class="product__title"
        >
          Product
        </h1>
        <div
          class="product__content"
        >
          <div
            class="product__content__skeleton"
          >
            <div
              class="product__content__skeleton__image"
            />
            <div
              class="product__content__skeleton__name"
            />
            <div
              class="product__content__skeleton__category"
            />
            <div
              class="product__content__skeleton__price"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
